ipaserver/install/ipa_idrange_fix.py

```diff
@@ -0,0 +1,1073 @@
+"""Tool to analyze and fix IPA ID ranges"""
+# Authors: Aleksandr Sharov <asharov@redhat.com>
+#
+# Copyright (C) 2024  Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import ldap
+
+from ipalib import api, errors
+from ipapython.admintool import AdminTool
+from ipapython.dn import DN
+from ipapython import ipautil
+
+logger = logging.getLogger(__name__)
+
+
+class IDRange:
+    """Class for ID range entity"""
+
+    def __init__(self, **kwargs):
+        self.last_id: int = None
+        self.last_base_rid: int = None
+        self.last_secondary_rid: int = None
+        self.name: str = kwargs.get('name')
+        self.size: int = kwargs.get('size')
+        self.first_id: int = kwargs.get('first_id')
+        self.base_rid: int = kwargs.get('base_rid')
+        self.secondary_base_rid: int = kwargs.get('secondary_base_rid')
+        self.type: str = kwargs.get('type')
+        self.suffix: str = kwargs.get('suffix')
+        self.dn: str = kwargs.get('dn')
+        if self.first_id is not None and self.size is not None:
+            self.count()
+
+    def count(self) -> None:
+        """Function to calculate last IDs for the range"""
+        self.last_id = self.first_id + self.size - 1
+        if self.type == "ipa-local":
+            self.last_base_rid = (
+                self.base_rid + self.size
+                if self.base_rid is not None
+                else None
+            )
+            self.last_secondary_rid = (
+                self.secondary_base_rid + self.size
+                if self.secondary_base_rid is not None
+                else None
+            )
+
+    def __repr__(self):
+        return f"IDRange(name='{self.name}', \
+type={self.type}, \
+size={self.size}, \
+first_id={self.first_id}, \
+base_rid={self.base_rid}, \
+secondary_base_rid={self.secondary_base_rid})"
```

Comment:
I guess the linters don't mind this indention but it looks a bit strange. I think I'd indent them at the same level as the rest of the statement.

Reply:
this was a single line, with a single pair of quotes, so it included tabs into output, hence 0 indentation. 

Fixed to a proper f string concatenation, and fixed indentation.